libwallet/src/account.rs

```diff
@@ -54,7 +55,7 @@ impl Account {
         self.pair.is_none()
     }
 
-    pub(crate) fn unlock(&mut self, root: &RootAccount) -> &Self {
+    pub(crate) fn unlock(mut self, root: &RootAccount) -> Self {
```

Comment:
⚠️  I believe we will have to kill `Account` or make it a trait that implies Signer, we are moving away form storing any private keys in the wallet code and only have that in the Vault implementations as some vaults won't give us access to any key pairs.